eng/jobs/steps/publish.yml

```diff
@@ -0,0 +1,53 @@
+parameters:
```

Comment:
From a high level view, the template name is a rather ambiguous IMO given there are other publish parts to the build.  Consider a more descriptive name.

Reply:
Agreed, I didn't have in mind how it looks outside the usage context.

Reply:
Changed to `upload-job-artifacts.yml`.